daemon/k8s_watcher.go

```diff
@@ -333,33 +336,48 @@ func (d *Daemon) addK8sNetworkPolicy(obj interface{}) {
 	scopedLog.Info("NetworkPolicy successfully added")
 }
 
-func (d *Daemon) updateK8sNetworkPolicy(oldObj interface{}, newObj interface{}) {
-	scopedLog := log.WithFields(log.Fields{
-		"obj.old": logfields.Repr(oldObj),
-		"obj.new": logfields.Repr(newObj),
-	})
-
-	k8sNP, ok := newObj.(*networkingv1.NetworkPolicy)
-	if ok {
-		scopedLog = scopedLog.WithField(logfields.K8sAPIVersion, k8sNP.APIVersion)
+func (d *Daemon) updateK8sNetworkPolicyV1(oldObj interface{}, newObj interface{}) {
+	// We don't need to deepcopy the object since we are creating a Cilium
+	// Network Policy rule with ParseNetworkPolicy below.
+	oldk8sNP, ok := oldObj.(*networkingv1.NetworkPolicy)
+	if !ok {
+		log.WithField(logfields.Object+".old", logfields.Repr(oldObj)).
+			Warn("Ignoring invalid k8s NetworkPolicy modification")
+		return
 	}
-
-	scopedLog.Debug("Modified policy")
-	d.addK8sNetworkPolicy(newObj)
+	newk8sNP, ok := newObj.(*networkingv1.NetworkPolicy)
+	if !ok {
+		log.WithField(logfields.Object+".new", logfields.Repr(newk8sNP)).
+			Warn("Ignoring invalid k8s NetworkPolicy modification")
+		return
+	}
+	log.WithFields(log.Fields{
+		logfields.K8sAPIVersion:                 oldk8sNP.TypeMeta.APIVersion,
+		logfields.K8sNetworkPolicyName + ".old": oldk8sNP.ObjectMeta.Name,
+		logfields.K8sNamespace + ".old":         oldk8sNP.ObjectMeta.Namespace,
+		logfields.K8sNetworkPolicyName + ".new": newk8sNP.ObjectMeta.Name,
```

Comment:
IMHO you should avoid creating new log field keys on the fly. When those logs are fed into an ELK stack, this will prevent correlating all the messages for this namespace. You should consistently use the same keys.
I'd suggest only renaming one pair of keys, e.g. keeping only the `+ ".old"`, or better log 2 messages with the different values.

This comment applies to the other places below where you create new keys like here.

Reply:
I'll do that later #2006